lib/realtime/tenants/authorization.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Realtime.Tenants.Authorization do
+  alias Plug.Conn
+  alias Phoenix.Socket
+
+  require Logger
+
+  def get_authorizations(%Socket{} = socket, db_conn, params) do
+    with {:ok, {:ok, permissions}} <- get_permissions_for_connection(db_conn, params) do
+      {:ok, Socket.assign(socket, :permissions, permissions)}
+    else
+      {:error, _} -> {:error, :unauthorized}
+    end
+  end
+
+  def get_authorizations(%Conn{} = conn, db_conn, params) do
+    with {:ok, {:ok, permissions}} <- get_permissions_for_connection(db_conn, params) do
+      {:ok, Conn.assign(conn, :permissions, permissions)}
+    else
+      {:error, _} -> {:error, :unauthorized}
+    end
+  end
+
+  defp get_permissions_for_connection(conn, params) do
+    %{
+      channel_name: channel_name,
+      headers: headers,
+      jwt: jwt,
+      claims: claims,
+      role: role
+    } = params
+
+    sub = Map.get(claims, :sub)
+    claims = Jason.encode!(claims)
+    headers = headers |> Map.new() |> Jason.encode!()
+
+    Postgrex.transaction(conn, fn conn ->
+      Postgrex.query!(
+        conn,
+        """
+        SELECT
+         set_config('role', $1, true),
+         set_config('realtime.channel_name', $2, true),
+         set_config('request.jwt.claim.role', $3, true),
+         set_config('request.jwt', $4, true),
+         set_config('request.jwt.claim.sub', $5, true),
```

Comment:
you should add "sub" for jwt validation
https://github.com/supabase/realtime/blob/d3549c4587e0357ae1f414485d89bc77dd89b82e/lib/realtime_web/channels/auth/channels_authorization.ex#L22-L29

Reply:
`sub` can be optional right? for example `anon` might not need a `sub` in their JWT